backend/composer/management/commands/ingest_anatomical_entities.py

```diff
@@ -1,64 +1,88 @@
 import csv
-
-from django.core.management.base import BaseCommand
-
-from composer.models import AnatomicalEntity
+import time
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.db.utils import IntegrityError
+from composer.models import AnatomicalEntity, Synonym
 
 URI = "o"
 NAME = "o_label"
 SYNONYM = "o_synonym"
+BULK_LIMIT = 100
 
 
 class Command(BaseCommand):
-    help = "Ingests Anatomical Entities CSV file(s)"
+    help = "Ingests Anatomical Entities CSV file(s), with robust error handling."
 
     def add_arguments(self, parser):
         parser.add_argument("csv_files", nargs="+", type=str)
+        parser.add_argument("--update_names", action='store_true',
+                            help="Update the name even if the entity already exists")
+        parser.add_argument("--show_complete_logs", action='store_true',
+                            help="Show detailed logs during processing")
+
+    def _process_anatomical_entity(self, name, ontology_uri, synonym, update_names, show_complete_logs, processed_uris,
+                                   synonym_accumulator):
+        try:
+            is_first_occurrence = ontology_uri not in processed_uris
 
-    def _create_ae(self, name, ontology_uri):
-        found = AnatomicalEntity.objects.filter(name__iexact=name).exists()
-        if not found:
-            return AnatomicalEntity(
-                name=name,
-                ontology_uri=ontology_uri
+            anatomical_entity, created = AnatomicalEntity.objects.get_or_create(
+                ontology_uri=ontology_uri,
+                defaults={"name": name},
             )
-        return None
-        # anatomical_entity, created = AnatomicalEntity.objects.get_or_create(
-        #     name__iexact=name,
-        #     defaults={"ontology_uri": ontology_uri, "name": name},
-        # )
-        # if created:
-        #     self.stdout.write(f"Anatomical Entity {name} created.")
-        #     anatomical_entity.save()
+            if not created and update_names and is_first_occurrence:
+                if anatomical_entity.name != name:
+                    anatomical_entity.name = name
+                    anatomical_entity.save()
+                    if show_complete_logs:
+                        self.stdout.write(
+                            self.style.SUCCESS(f"Updated {anatomical_entity.ontology_uri} name to {name}."))
 
+            processed_uris.add(ontology_uri)
```

Comment:
remark: @afonsobspinto imo we don't need to track processed uris

Reply:
See https://github.com/MetaCell/sckan-composer/pull/249#discussion_r1528814178 for the explanation on why I believe we need it